### PR TITLE
Moved proto to be only proto files and sources to utils

### PR DESCRIPTION
**Pull request by Yuhanun**
Waiting for other repos.

### CODE REVIEW

**Review by rolfvdhulst — CHANGES_REQUESTED** [review]
The cleanest way to do this is to actually keep two things in the Roboteam_Proto repository:
1. The proto files
2. A CMakelists that compiles all of the proto files into a c++ code/ target/library (e.g. roboteam_proto).

Then for the networking in utils you can simply do something like this:
target_link_libraries(networking PUBLIC roboteam_proto).

Note this also helps keep general compile times down as it reduces the need to search in the network headers everytime you link roboteam_proto, and it's a lot more modular / easy to work with.

**Comment by Yuhanun** [comment]
> The cleanest way to do this is to actually keep two things in the Roboteam_Proto repository:
> 
> 1. The proto files
> 2. A CMakelists that compiles all of the proto files into a c++ code/ target/library (e.g. roboteam_proto).
> 
> Then for the networking in utils you can simply do something like this:
> target_link_libraries(networking PUBLIC roboteam_proto).
> 
> Note this also helps keep general compile times down as it reduces the need to search in the network headers everytime you link roboteam_proto, and it's a lot more modular / easy to work with.

I actually think the way it is right now is how it should be. You shouldn't need a CMakelists.txt in a submodule on a web-UI. I think this is much better, because you can literally just add the proto files as a submodule for anything. 

The whole point of this was so that you don't have non-js stuff in the UI, were we to add the cmakelists there it kinda defeats that purpose.

**Comment by rolfvdhulst** [comment]
> I actually think the way it is right now is how it should be. You shouldn't need a CMakelists.txt in a submodule on a web-UI. I think this is much better, because you can literally just add the proto files as a submodule for anything.
> 
> The whole point of this was so that you don't have non-js stuff in the UI, were we to add the cmakelists there it kinda defeats that purpose.

Why would you need the CMakelists.txt for the interface? Yes, it would be present in the webrepository, but I'm not saying you should try to do the Protobuf->Javascript conversion using CMake (definitely don't do that).

The point I'm making is that for the c++ parts of our system, having a seperate target for the proto files and for the networking files is nice. I personally think that outweighs having a useless CMakeLists.txt in the directory when you're doing interface development. Now you have to do ugly things with directories/path dependencies in Roboteam_utils to get the proto files to work nicely in the rest of the system, because to compile them you first need to find the repository.
 Compile it as a c++ library where the proto files are defined and it's modular and easier to work with/change in the future. With the current solution you're proposing, also every time someone wants to add/remove a proto file, they have to make a pull request in two repositories instead of one.

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,118 +1,61 @@
...
         PUBLIC ${PROTOBUF_LIBRARIES}
-        # since Publishers/subscribers are templates these 2 libraries need to be PUBLIC
-        PUBLIC ${ZeroMQ_LIBRARIES}
-        PUBLIC ${ZMQPP_LIBRARIES}
         )
 
 target_include_directories(roboteam_proto
+        # Proto headers
         PUBLIC ${PROTOBUF_INCLUDE_DIRS}
+        # This points to cmake-build-debug/utils/include/utils/networking
```
Comment (rolfvdhulst) [inline CMakeLists.txt]: This comment is outdated.

**Review by rolfvdhulst — APPROVED** [review]
LGTM, look at the one comment though.
Also, maybe consider deleting large parts of the readme, it's horribly outdated.

File: CMakeLists.txt (hunk 1 above)
Comment (Yuhanun) [inline CMakeLists.txt]:
```suggestion
        # This points to cmake-build-debug/proto/include/proto
```

**Pull request merged**